Code/Framework/AzCore/AzCore/Component/ComponentApplication.cpp

```diff
@@ -587,6 +537,172 @@ namespace AZ
+        {
+            return;
+        }
+
+        // Append the build configuration(debug, release, profile) to the metrics filename
+        if (AZStd::string_view buildConfig(AZ_BUILD_CONFIGURATION_TYPE); !buildConfig.empty())
+        {
+            uniqueFilenameSuffix.Native() += AZ::IO::FixedMaxPathString::format(".%.*s", AZ_STRING_ARG(buildConfig));
+        }
+
+        // Retrieve a filename safe timestamp with milliseconds precision
+        if (AZ::Date::Iso8601TimestampString timestampMilli;
+            AZ::Date::GetFilenameCompatibleFormatNowWithMilliseconds(timestampMilli))
+        {
+            uniqueFilenameSuffix.Native() += AZ::IO::FixedMaxPathString::format(".%s", timestampMilli.c_str());
+        }
+        // Append .json extension
+        uniqueFilenameSuffix.Native() += ".json";
+
+        // Append the relative file name portion to the <project-root>/user directory
+        auto metricsFilePath = (AZ::IO::FixedMaxPath{ AZ::Utils::GetProjectUserPath(m_settingsRegistry.get()) }
+            / uniqueFilenameSuffix).LexicallyNormal();
+
+        // Open up the metrics file in write mode and truncate the contents if it exist(it shouldn't since a millisecond
+        // is being used
+        constexpr AZ::IO::OpenMode openMode = AZ::IO::OpenMode::ModeWrite | AZ::IO::OpenMode::ModeCreatePath;
+        if (auto fileStream = AZStd::make_unique<AZ::IO::SystemFileStream>(metricsFilePath.c_str(), openMode);
+            fileStream != nullptr && fileStream->IsOpen())
+        {
+            // Configure core event logger with the name of "Core"
+            AZ::Metrics::JsonTraceLoggerEventConfig config{ "Core" };
+            auto coreEventLogger = AZStd::make_unique<AZ::Metrics::JsonTraceEventLogger>(AZStd::move(fileStream), config);
+            m_eventLoggerFactory->RegisterEventLogger(AZ::Metrics::CoreEventLoggerId, AZStd::move(coreEventLogger));
+        }
+        else
+        {
+            AZ_Error("ComponentApplication", false, R"(unable to open core metrics with with path "%s")",
+                metricsFilePath.c_str());
+        }
+
+        // Record metrics every X microseconds based on the /O3DE/Metrics/Core/RecordRateMicroseconds setting
+        // or every 10 secondsif not supplied
+        m_recordMetricsOnTickCallback = [&registry = *m_settingsRegistry.get(),
+            lastRecordTime = AZStd::chrono::steady_clock::now()]
+            (AZStd::chrono::steady_clock::time_point monotonicTime) mutable -> bool
+        {
+            // Retrieve the record rate setting from the Setting Registry
+            using namespace AZStd::chrono_literals;
+            AZStd::chrono::microseconds recordTickMicroseconds = 10s;
+            if (AZ::s64 recordRateMicrosecondValue;
+                registry.Get(recordRateMicrosecondValue, AZ::Metrics::CoreMetricsRecordRateMicrosecondsKey))
+            {
+                recordTickMicroseconds = AZStd::chrono::microseconds(recordRateMicrosecondValue);
+            }
```

Comment:
minor:
Is it possible that registry value changes during runtime? If not, maybe `recordTickMicroseconds` can be moved out of this lambda and just captured in.

Reply:
It is possible to update the record rate during run time via the SettingsRegistryInterface::Set function in C++ or through scripting.